src/v/pandaproxy/schema_registry/service.cc

```diff
@@ -202,18 +202,37 @@ ss::future<> service::configure() {
       _ctx.smp_sg, [has_ephemeral_credentials](service& s) {
           s._has_ephemeral_credentials = has_ephemeral_credentials;
       });
-    co_await _controller->get_security_frontend().local().create_acls(
-      {security::acl_binding{
-        security::resource_pattern{
-          security::resource_type::topic,
-          model::schema_registry_internal_tp.topic,
-          security::pattern_type::literal},
-        security::acl_entry{
-          principal,
-          security::acl_host::wildcard_host(),
-          security::acl_operation::all,
-          security::acl_permission::allow}}},
-      5s);
+
+    std::vector<cluster::errc> err_vec;
+    // This could be any non-success errc
+    err_vec.push_back(cluster::errc::timeout);
+
+    while (err_vec[0] != cluster::errc::success) {
+        err_vec
+          = co_await _controller->get_security_frontend().local().create_acls(
+            {security::acl_binding{
+              security::resource_pattern{
+                security::resource_type::topic,
+                model::schema_registry_internal_tp.topic,
+                security::pattern_type::literal},
+              security::acl_entry{
+                principal,
+                security::acl_host::wildcard_host(),
+                security::acl_operation::all,
+                security::acl_permission::allow}}},
+            5s);
+
+        if (err_vec[0] != cluster::errc::success) {
+            vlog(
+              plog.info,
+              "Failed to create ACLs, User {}, err {} - {}",
+              principal,
+              err_vec[0],
+              cluster::make_error_code(err_vec[0]).message());
+            co_await ss::sleep(5s);
```

Comment:
#11141 was failing because the create_acls request here was silently failing either because the serving node was not raft0 leader or because the controller topic did not have a leader yet. Here are some examples of the logs:
```
Waiting on controller leader ...

INFO  2023-06-20 13:03:35,004 [shard 0] pandaproxy - service.cc:225 - Failed to create ACLs, User {type ephemeral user name __schema_registry}, err cluster::errc:9 - Currently there is no leader controller elected in the cluster


Waiting for raft-0 leader ...

INFO  2023-06-20 13:01:51,273 [shard 0] pandaproxy - service.cc:225 - Failed to create ACLs, User {type ephemeral user name __schema_registry}, err cluster::errc:5 - This node is not raft-0 leader. i.e is not leader controller
```

My local tests pass when I add in a long enough sleep between retries for create_acls; however, I do not believe injecting a long sleep like this is OK. Therefore, I made this comment to collect feedback.

I believe this indicates that SR startup order needs to change or perhaps we should create ACLs for the ephemeral user in a different location. Finally, we did not get the failure in #11141 until now because SR startup was changed with the new addition of schema_id_validation.

Reply:
Yeh there are some good options for this.

1. What Mike suggested
2. If failure occurs at start-up, try to complete it in the background
3. Retry without a 5s fixed timeout--try 10ms, 100ms, 1sec, ... since chances are high that it's just an ephemeral issue

Reply:
Chatted with @NyaliaLui about why #1 isn't tenable solution.  The issue is that when this call fails, there is no way for SR to access the __schema topic.  Is there a way to know when initial controller leadership election is done?

Reply:
> 3. Retry without a 5s fixed timeout--try 10ms, 100ms, 1sec, ... since chances are high that it's just an ephemeral issue

This is a good idea; however, there is more info that is hard to explain in text but I'll give it a try.

A small timeout (10ms or less) is sufficient for `create_acls` to eventually succeed. Unfortunately, the tests still fail unless I add a larger timeout (5s or more). IMO, this indicates that there are two problems with SR startup on `dev` at the moment:
1. If `create_acls` fails, then that Kafka broker does not have ACLs for the ephemeral principal.
2. There is some other dependency to SR startup that is necessary for ACLs checks to succeed. Adding a 1s or greater timeout allows enough time for this unknown dependency to stabilize.

Adding the retry loop in this PR addresses point (1) but I'm still searching for this unknown dependency in point (2).

Reply:
how we doing here? should the three of us have a quick chat to converge on a solution?

Reply:
@NyaliaLui and @BenPope and I had a chat about this and we may have a path forward (though I just got an invite to a meeting with replication)

Reply:
+1